godot-macros/src/lib.rs

```diff
@@ -455,14 +456,56 @@ pub fn derive_from_variant(input: TokenStream) -> TokenStream {
     translate(input, derive_from_variant::transform)
 }
 
-/// Derive macro for [Property](godot::Property)
+/// Derive macro for [Property](godot::bind::property::Property) on enums.
 ///
-/// TODO: document
+/// Currently has some tight requirements which are expected to be softened as implementation expands:
+/// - Only works for enums, structs aren't supported by this derive macro at the moment.
+/// - The enum must have an explicit `#[repr(u*/i*)]` type.
+///     - This will likely stay this way, since `isize`, the default repr type, is not a concept in Godot.
+/// - The enum variants must not have any fields - currently only unit variants are supported.
+/// - The enum variants must have explicit discriminants, that is, e.g. `A = 2`, not just `A`
+///     - This will likely stay this way in order to keep proc-macro dependencies lighter.
+///
+/// Example :
+///
+/// ```ignore
```

Comment:
Could you use _no_run_ instead of _ignore_?

That way, the code is still compiled. Use the `#` placeholder to hide imports like the prelude.